qiskit_ibm_runtime/runtime_job_v2.py

```diff
@@ -0,0 +1,788 @@
+        """
+        try:
+            result_url_json = json.loads(response)
+            if "url" in result_url_json:
+                url = result_url_json["url"]
+                result_response = requests.get(url, timeout=10)
+                return result_response.text
+            return response
+        except json.JSONDecodeError:
+            return response
+
+    def interim_results(self, decoder: Optional[Type[ResultDecoder]] = None) -> Any:
+        """Return the interim results of the job.
+
+        Args:
+            decoder: A :class:`ResultDecoder` subclass used to decode interim results.
+
+        Returns:
+            Runtime job interim results.
+
+        Raises:
+            RuntimeJobFailureError: If the job failed.
+        """
+        if not self._final_interim_results:
+            _decoder = decoder or self._interim_result_decoder
+            interim_results_raw = self._api_client.job_interim_results(job_id=self.job_id())
+            self._interim_results = _decoder.decode(interim_results_raw)
+            if self.status() in JOB_FINAL_STATES:
+                self._final_interim_results = True
+        return self._interim_results
+
+    def result(  # pylint: disable=arguments-differ
+        self,
+        timeout: Optional[float] = None,
+        decoder: Optional[Type[ResultDecoder]] = None,
+    ) -> Any:
+        """Return the results of the job.
+
+        Args:
+            timeout: Number of seconds to wait for job.
+            decoder: A :class:`ResultDecoder` subclass used to decode job results.
+
+        Returns:
+            Runtime job result.
+
+        Raises:
+            RuntimeJobFailureError: If the job failed.
+            RuntimeJobMaxTimeoutError: If the job does not complete within given timeout.
+            RuntimeInvalidStateError: If the job was cancelled, and attempting to retrieve result.
+        """
+        _decoder = decoder or self._final_result_decoder
+        self.wait_for_final_state(timeout=timeout)
+        if self._status == "ERROR":
+            error_message = self._reason if self._reason else self._error_message
+            if self._reason == "RAN TOO LONG":
+                raise RuntimeJobMaxTimeoutError(error_message)
+            raise RuntimeJobFailureError(f"Unable to retrieve job result. {error_message}")
+        if self._status == "CANCELLED":
+            raise RuntimeInvalidStateError(
+                "Unable to retrieve result for job {}. " "Job was cancelled.".format(self.job_id())
+            )
+
+        result_raw = self._download_external_result(
+            self._api_client.job_results(job_id=self.job_id())
+        )
+        # TODO: Remove getting/setting version once it's in result metadata
+        if _decoder.__name__ == EstimatorResultDecoder.__name__:
+            if not self._version:
+                self._version = self.inputs.get("version", 1)
+
+        return _decoder.decode(result_raw) if result_raw else None  # type: ignore
```

Comment:
I don't see the version in the result metadata, but removed the `version_param` object because it's not accepted by `estimator_result_decoder` anymore